lib/posix/pthread.c

```diff
@@ -410,54 +564,52 @@ void pthread_exit(void *retval)
 		}
 	}
 
-	pthread_mutex_unlock(&self->state_lock);
-	pthread_mutex_destroy(&self->state_lock);
-
-	pthread_cond_destroy(&self->state_cond);
-
-	k_thread_abort((k_tid_t)self);
+	posix_thread_finalize(self, retval);
+	CODE_UNREACHABLE;
 }
 
 /**
  * @brief Wait for a thread termination.
  *
  * See IEEE 1003.1
  */
-int pthread_join(pthread_t thread, void **status)
+int pthread_join(pthread_t th, void **status)
 {
-	struct posix_thread *pthread = to_posix_thread(thread);
-	int ret = 0;
+	int err;
+	int qid;
+	int detachstate;
+	k_spinlock_key_t key;
+	struct posix_thread *t;
```

Comment:
would be nice for the `struct posix_thread *` variable in various functions to use the same name, along with other variables.

Reply:
true. I'll make that change.